api/src/main/java/org/apache/iceberg/catalog/Catalog.java

```diff
@@ -97,6 +98,42 @@ default Table createTable(
     return createTable(identifier, schema, PartitionSpec.unpartitioned(), null, null);
   }
 
+  /**
+   * Start a transaction to create a table.
+   *
+   * @param identifier a table identifier
+   * @param schema a schema
+   * @param spec a partition spec
+   * @param location a location for the table; leave null if unspecified
+   * @param properties a string map of table properties
+   * @return a {@link Transaction} to create the table
+   * @throws AlreadyExistsException if the table already exists
+   */
+  Transaction newCreateTableTransaction(
```

Comment:
I am a bit concerned about the number of methods we will have in `Catalog`. So, I did not include overloaded helper methods as we have for create table statements. Let me know what other people think.

Reply:
I'd rather keep them in, but it should be fine this way. We can always add them later.

Reply:
I'll add them